install/helm/agones/templates/processor.yaml

```diff
@@ -0,0 +1,178 @@
+# Copyright 2025 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- $featureGates := include "agones.featureGates" . | fromYaml }}
+{{- if $featureGates.ProcessorAllocator }}
+apiVersion: apps/v1
```

Comment:
I'm assuming documentation comes last? (i.e. we don't want to include the Helm config docs now -- we'll wait until everything is all worked out and do it all at once?) Not against the idea, just wanted to check.

Reply:
Yep exactly, I'll write the documentation at the end, I've did some diagrams at the issue level in the comments part: https://github.com/googleforgames/agones/issues/4190 

I could probably start writing the doc in another PR while implementing the allocator / processor / extension, this would give more feedback and a better understanding of the big picture, also ensure we don't miss anything ? WDYT ? 

Even though the doc would only be merged once moved to Alpha I guess ?

Reply:
I'm not super fussed on the documentation part - the config files are enough for the moment, and if we find something along the way to change, you can just do one set of docs at the end rather than constantly updating 👍🏻